sematic/log_reader.py

```diff
@@ -340,16 +341,33 @@ def load_log_lines(
             reverse=reverse,
             **cursor_kwargs,  # type: ignore
         )
+    else:
+        log_line_result = _load_non_inline_logs(
+            run_id=run_id,
+            still_running=still_running,
+            max_lines=max_lines,
+            filter_strings=filter_strings,
+            default_log_info_message=default_log_info_message,
+            reverse=reverse,
+            **cursor_kwargs,  # type: ignore
+        )
 
-    return _load_non_inline_logs(
-        run_id=run_id,
-        still_running=still_running,
-        max_lines=max_lines,
-        filter_strings=filter_strings,
-        default_log_info_message=default_log_info_message,
-        reverse=reverse,
-        **cursor_kwargs,  # type: ignore
-    )
+    # when no cursor is set, it means that we are doing the initial pull
+    # and we can determine if we can continue in either direction
+    # this overrides the value from the `log_line_result`.
+    if reverse_cursor_token is None and forward_cursor_token is None:
+        if reverse is True:
+            # can_continue_forward will be False
+            # if the run has reached to a terminal state
+            log_line_result = replace(
+                log_line_result,
+                can_continue_forward=run.future_state
+                not in FutureState.terminal_state_strings(),
+            )
+        else:
```

Comment:
```suggestion
        elif forward_cursor_token is None:
```
similar reasoning to above: you can un-nest this from the if, and it's more correct to ignore the other cursor here anyway.

Reply:
Similarly, I guess this will take care of `reverse=None, reverse_cursor_token=something with forward_cursor_token=None?` it isn't the business of this logic, so I'd better exclude that.